### PR TITLE
wifi: Add a new NW selection criterion

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -247,4 +247,22 @@ endif
...
+	help
+		Select the network selection method for the supplicant.
+
+config WPA_SUPP_NW_SEL_THROUGHPUT
+	bool "Throughput based network selection"
+	help
+	  Select the network based on throughput.
+
+config WPA_SUPP_NW_SEL_RELIABILITY
+	bool "Reliability based network selection"
```
Comment (markaj-nordic) [inline modules/hostap/Kconfig:263]: What is the reliability indicator in this case?

**Review by markaj-nordic — COMMENTED** [review]
What would be the exemplary use case? Scanning with multiple networks with the same SSID?

File: modules/hostap/Kconfig (hunk 1 above)
Comment (krish2718) [reply modules/hostap/Kconfig:263]: Reliable connection i.e., no disconnections. E.g., choosing a far network over a near network.

**Comment by krish2718** [comment]
> What would be the exemplary use case? Scanning with multiple networks with the same SSID?

Yeah, Enterprise or mesh.

**Review by jukkar — APPROVED** [review]

**Pull request merged**